processor/src/it/java/com/linecorp/decaton/processor/RetryQueueingTest.java

```diff
@@ -99,4 +134,46 @@ public void testRetryQueuing() throws Exception {
                 .build()
                 .run();
     }
+
+    /*
+     * This test tries to re-produce delivery-loss due to https://github.com/line/decaton/issues/101 by:
+     *   1. Retry all tasks once immediately
+     *   2. Process retried tasks in "slow" pace
+     *     - To simulate the situation like "Retry tasks spikes (e.g. due to downstream storage failure),
+     *       but processing throughput is not sufficient", which is typical situation that causes
+     *       message delivery loss due to decaton#101.
+     */
+    @Test(timeout = 60000)
+    public void testRetryQueuingExtractingWithDefaultMeta() throws Exception {
+        // LogManager waits to start clean-up process until InitialTaskDelayMs (30sec) elapses.
+        // https://github.com/apache/kafka/blob/2.4.0/core/src/main/scala/kafka/log/LogManager.scala#L70
+        Thread.sleep(30000L);
+
+        ProcessorTestSuite
+                .builder(rule)
+                .numTasks(10)
+                .propertySupplier(StaticPropertySupplier.of(
+                        // Configure max pending records to 1 to pause fetches easily
+                        Property.ofStatic(ProcessorProperties.CONFIG_MAX_PENDING_RECORDS, 1),
+                        Property.ofStatic(ProcessorProperties.CONFIG_PARTITION_CONCURRENCY, 1)
+                ))
+                .configureProcessorsBuilder(builder -> builder.thenProcess((ctx, task) -> {
+                    if (ctx.metadata().retryCount() == 0) {
+                        ctx.retry();
+                    } else {
+                        Thread.sleep(1000L);
```

Comment:
Assuming the purpose of this sleep is to let broker enough time to run retention check of log segments, is it fine to apply it just for the first record so the overall test duration won't get long?

Reply:
btw this needs to be sufficient larger than https://github.com/line/decaton/pull/102/files#diff-50a812e0953575d473c752613b37560bdd78696bae7e051019e48e21d278029eR60 this right? How about definining a constant and use it so the relationship will be clearer?

Reply:
> is it fine to apply it just for the first record

Sounds enough. Let me fix